src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,224 @@
-import spray.http._
+import org.broadinstitute.dsde.firecloud.service.TSVWriterActor._
+import org.broadinstitute.dsde.firecloud.utils.StreamingActor.{FirstChunk, NextChunk}
+import org.broadinstitute.dsde.firecloud.utils.{StreamingActor, TSVFormatter}
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudConfig, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.rawls.model._
+import org.slf4j.{Logger, LoggerFactory}
+import spray.http.{ContentTypes, _}
+import spray.routing.RequestContext
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.Success
+import scala.language.postfixOps
+
+// JSON Serialization Support
+import spray.json._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+
 
 object ExportEntitiesByTypeActor {
   sealed trait ExportEntitiesByTypeMessage
-  case class ExportEntities(workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
+  case class ExportEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
 
   def props(exportEntitiesByTypeConstructor: UserInfo => ExportEntitiesByTypeActor, userInfo: UserInfo): Props = {
     Props(exportEntitiesByTypeConstructor(userInfo))
   }
 
   def constructor(app: Application)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
-    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo)
+    new ExportEntitiesByTypeActor(app.rawlsDAO, app.googleServicesDAO, userInfo)
 }
 
-class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val googleDAO: GoogleServicesDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+  // Requires its own actor context to work with downstream actors: TSVWriterActor and StreamingActor
+  def actorRefFactory: ActorContext = context
   override def receive: Receive = {
-    case ExportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) => exportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) pipeTo sender
+    case ExportEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) => streamEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) pipeTo sender
   }
 }
 
 trait ExportEntitiesByType extends FireCloudRequestBuilding {
   val rawlsDAO: RawlsDAO
+  val googleDAO: GoogleServicesDAO
   implicit val userInfo: UserInfo
   implicit protected val executionContext: ExecutionContext
+  implicit def actorRefFactory: ActorRefFactory
```

Comment:
I think I figured this out what your intention is. In the existing style, we have actor classes that are super thin and then any real work is done in a trait. In this case, that doesn't make as much sense since we have so much actor-based work to do. Pulling all of the helper methods into the actor class cleans this up a lot.